pkg/dependency/dag.go

```diff
@@ -28,6 +28,8 @@ type SinksPerSource struct {
 type Source struct {
 	// ObjectRef refers to the object this source come from.
 	ObjectRef *corev1.ObjectReference
+
+	Matchers []v1alpha2.ConditionRequirement
```

Comment:
Would you consider not using ObjectReference and just spell out all the needed fields?  Not sure why we need to borrow an object reference

Reply:
Why don't you want borrow object reference here? They're exactly same